packages/tx/src/baseTransaction.ts

```diff
@@ -90,11 +91,26 @@ export abstract class BaseTransaction<TransactionObject> {
     const vB = toBuffer(v === '' ? '0x' : v)
     const rB = toBuffer(r === '' ? '0x' : r)
     const sB = toBuffer(s === '' ? '0x' : s)
+    const nonceB = toBuffer(nonce === '' ? '0x' : nonce)
+    const gasLimitB = toBuffer(gasLimit === '' ? '0x' : gasLimit)
+    const valueB = toBuffer(value === '' ? '0x' : value)
 
-    this.nonce = new BN(toBuffer(nonce === '' ? '0x' : nonce))
-    this.gasLimit = new BN(toBuffer(gasLimit === '' ? '0x' : gasLimit))
+    if (gasLimitB.length > 8) {
+      // Geth limits gasLimit to Uint64 so gas cannot be more than 2^64-1
+      throw new Error('gasLimit cannot be more than 2^64-1')
+    }
+
+    this.nonce = new BN(nonceB)
+
+    if (this.nonce.gte(MAX_UINT64)) {
+      // Implements EIP-2681 which limits nonce to 2^64-1
+      // https://eips.ethereum.org/EIPS/eip-2681
+      throw new Error('nonce cannot be more than 2^64-1')
+    }
```

Comment:
Sorry for being a bit picky here - just live-thinking myself. I am just wondering: since the `BN.js` library is somewhat on its last iterations and we will at some point replace with `BigInt` *and* we have this other check possibility already in here (which also makes the code a bit inconsistent):

Maybe (or - getting a a stronger opinion while writing - likely 😋) we just want to skip this `MAX_UINT64` and also remove from `Util` and not introduce a new `BN.js` related constant for this one single first use case and rather do a byte length check here as well as being done for `gasLimitB`.

Reply:
Okay, I'm good with it either way.  I like the idea of just checking the bytes myself but there was concern in earlier reviews about converting BNs to buffers causing additional performance hits (right @ryanio?) or something to that effect.

Reply:
Yeah sure just check the nonceB like the others to match 👍

Reply:
So...doing the bytelength comparison actually doesn't work because we have to check not only 2^64 -1 but 2^64-2 per EIP 2681 - "The reason a transaction with nonce 2^64-1 is invalid, because otherwise after inclusion the sender account’s nonce would exceed 2^64-1."  So, I'm going to have to use the new BN either way.

Reply:
Ok, then just keep the new BN constant in Util, just realized that there are already two BN constants in so we have to deal with this at some point anyhow and really makes no difference if there are 2 or 3 I guess.